### DISCUSSION TITLE
automatic user timezone cast

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#36796] [Ideas]
Thinking it would be nice if 

- dates automatically cast to the auth user's timezone.
- a timezone column default on users

**Comment by selcukcukur** [comment]
You can provide this during user registration anyway, I could not understand exactly what you wanted.

**Reply by dillingham** [reply]
@selcukcukur I think it might be nice if model's timestamps casted to date would automatically be in the active user model's timezone since an app is centered around the user vs converting on the frontend or anywhere else. Same with DB.

**Reply by dillingham** [reply]
This is probably a bad idea lol

**Reply by selcukcukur** [reply]
this will require a lot of extra query and editing. Because many arrangements will be required, such as the date information of the transactions that are adjusted according to the user time zone, and the arrangements for the time zone difference. Therefore, it will always be more logical to use the time zone of the system and translate it when necessary, or to provide users with a time zone option and to make comparisons between server time and user time, otherwise it will create unnecessary load.

**Reply by selcukcukur** [reply]
@dillingham In short, the system will use its own time zone and a comparison will be made for the user, if necessary, and the necessary action will be taken according to the difference. Otherwise, it would be a bad idea to have this procedure done all the time.
